### ISSUE TITLE
autoCenter not working...

### ISSUE THREAD

**Issue by doorman02** [swimlane/ngx-graph#191]
**Describe the bug**
When using the autoCenter input attribute I get the the following error 

```
Error: <g> attribute transform: Expected number, "matrix(1,0,0,1,NaN,NaN)".
push../node_modules/@angular/platform-browser/fesm5/platform-browser.js.DefaultDomRenderer2.setAttribute @ platform-browser.js:1095
push../node_modules/@angular/platform-browser/fesm5/animations.js.BaseAnimationRenderer.setAttribute @ animations.js:234
push../node_modules/@angular/core/fesm5/core.js.DebugRenderer2.setAttribute @ core.js:24305
setElementAttribute @ core.js:21072
checkAndUpdateElementValue @ core.js:21048
checkAndUpdateElementInline @ core.js:21008
checkAndUpdateNodeInline @ core.js:23359
checkAndUpdateNode @ core.js:23325
debugCheckAndUpdateNode @ core.js:23959
debugCheckRenderNodeFn @ core.js:23945
(anonymous) @ GraphComponent.html:12
debugUpdateRenderer @ core.js:23937
checkAndUpdateView @ core.js:23312
callViewAction @ core.js:23548
execEmbeddedViewsAction @ core.js:23511
checkAndUpdateView @ core.js:23308
callViewAction @ core.js:23548
execComponentViewsAction @ core.js:23490
checkAndUpdateView @ core.js:23313
callViewAction @ core.js:23548
execEmbeddedViewsAction @ core.js:23511
checkAndUpdateView @ core.js:23308
callViewAction @ core.js:23548
execComponentViewsAction @ core.js:23490
checkAndUpdateView @ core.js:23313
callWithDebugContext @ core.js:24177
debugCheckAndUpdateView @ core.js:23879
push../node_modules/@angular/core/fesm5/core.js.ViewRef_.detectChanges @ core.js:21688
(anonymous) @ core.js:18119
push../node_modules/@angular/core/fesm5/core.js.ApplicationRef.tick @ core.js:18119
(anonymous) @ core.js:18010
push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke @ zone.js:391
onInvoke @ core.js:17299
push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke @ zone.js:390
push../node_modules/zone.js/dist/zone.js.Zone.run @ zone.js:150
push../node_modules/@angular/core/fesm5/core.js.NgZone.run @ core.js:17213
next @ core.js:18010
schedulerFn @ core.js:13515
push../node_modules/rxjs/_esm5/internal/Subscriber.js.SafeSubscriber.__tryOrUnsub @ Subscriber.js:194
push../node_modules/rxjs/_esm5/internal/Subscriber.js.SafeSubscriber.next @ Subscriber.js:132
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber._next @ Subscriber.js:76
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:53
push../node_modules/rxjs/_esm5/internal/Subject.js.Subject.next @ Subject.js:47
push../node_modules/@angular/core/fesm5/core.js.EventEmitter.emit @ core.js:13499
checkStable @ core.js:17268
onLeave @ core.js:17335
onInvokeTask @ core.js:17293
push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask @ zone.js:422
push../node_modules/zone.js/dist/zone.js.Zone.runTask @ zone.js:195
push../node_modules/zone.js/dist/zone.js.ZoneTask.invokeTask @ zone.js:498
ZoneTask.invoke @ zone.js:487
timer @ zone.js:2281
requestAnimationFrame (async)
scheduleTask @ zone.js:2302
push../node_modules/zone.js/dist/zone.js.ZoneDelegate.scheduleTask @ zone.js:410
onScheduleTask @ zone.js:301
push../node_modules/zone.js/dist/zone.js.ZoneDelegate.scheduleTask @ zone.js:404
push../node_modules/zone.js/dist/zone.js.Zone.scheduleTask @ zone.js:238
push../node_modules/zone.js/dist/zone.js.Zone.scheduleMacroTask @ zone.js:261
scheduleMacroTaskWithCurrentZone @ zone.js:1245
(anonymous) @ zone.js:2317
proto.<computed> @ zone.js:1569
push../node_modules/@swimlane/ngx-graph/fesm5/swimlane-ngx-graph.js.GraphComponent.createGraph @ swimlane-ngx-graph.js:1494
(anonymous) @ swimlane-ngx-graph.js:1433
push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke @ zone.js:391
onInvoke @ core.js:17299
push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke @ zone.js:390
push../node_modules/zone.js/dist/zone.js.Zone.run @ zone.js:150
push../node_modules/@angular/core/fesm5/core.js.NgZone.run @ core.js:17213
push../node_modules/@swimlane/ngx-graph/fesm5/swimlane-ngx-graph.js.GraphComponent.update @ swimlane-ngx-graph.js:1423
push../node_modules/@swimlane/ngx-graph/fesm5/swimlane-ngx-graph.js.GraphComponent.setLayoutSettings @ swimlane-ngx-graph.js:1334
push../node_modules/@swimlane/ngx-graph/fesm5/swimlane-ngx-graph.js.GraphComponent.setLayout @ swimlane-ngx-graph.js:1320
push../node_modules/@swimlane/ngx-graph/fesm5/swimlane-ngx-graph.js.GraphComponent.ngOnChanges @ swimlane-ngx-graph.js:1299
checkAndUpdateDirectiveDynamic @ core.js:22118
checkAndUpdateNodeDynamic @ core.js:23379
checkAndUpdateNode @ core.js:23328
debugCheckAndUpdateNode @ core.js:23959
debugCheckDirectivesFn @ core.js:23919
(anonymous) @ AppComponent.html:7
debugUpdateDirectives @ core.js:23911
checkAndUpdateView @ core.js:23307
callViewAction @ core.js:23548
execEmbeddedViewsAction @ core.js:23511
checkAndUpdateView @ core.js:23308
callViewAction @ core.js:23548
execComponentViewsAction @ core.js:23490
checkAndUpdateView @ core.js:23313
callWithDebugContext @ core.js:24177
debugCheckAndUpdateView @ core.js:23879
push../node_modules/@angular/core/fesm5/core.js.ViewRef_.detectChanges @ core.js:21688
(anonymous) @ core.js:18119
push../node_modules/@angular/core/fesm5/core.js.ApplicationRef.tick @ core.js:18119
push../node_modules/@angular/core/fesm5/core.js.ApplicationRef._loadComponent @ core.js:18153
push../node_modules/@angular/core/fesm5/core.js.ApplicationRef.bootstrap @ core.js:18095
(anonymous) @ core.js:17808
push../node_modules/@angular/core/fesm5/core.js.PlatformRef._moduleDoBootstrap @ core.js:17808
(anonymous) @ core.js:17776
push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke @ zone.js:391
onInvoke @ core.js:17299
push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke @ zone.js:390
push../node_modules/zone.js/dist/zone.js.Zone.run @ zone.js:150
(anonymous) @ zone.js:889
push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask @ zone.js:423
onInvokeTask @ core.js:17290
push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask @ zone.js:422
push../node_modules/zone.js/dist/zone.js.Zone.runTask @ zone.js:195
drainMicroTaskQueue @ zone.js:601
Promise.then (async)
scheduleMicroTask @ zone.js:584
push../node_modules/zone.js/dist/zone.js.ZoneDelegate.scheduleTask @ zone.js:413
push../node_modules/zone.js/dist/zone.js.Zone.scheduleTask @ zone.js:238
push../node_modules/zone.js/dist/zone.js.Zone.scheduleMicroTask @ zone.js:258
scheduleResolveOrReject @ zone.js:879
ZoneAwarePromise.then @ zone.js:1012
push../node_modules/@angular/core/fesm5/core.js.PlatformRef.bootstrapModule @ core.js:17803
./src/main.ts @ main.ts:11
__webpack_require__ @ bootstrap:76
0 @ main.ts:12
__webpack_require__ @ bootstrap:76
checkDeferredModules @ bootstrap:43
webpackJsonpCallback @ bootstrap:30
(anonymous) @ main.js:1
Show 76 more frames
```

**To Reproduce**
Open the graph with the autoCenter attribute

**Expected behavior**
With autocenter set the graph should be centralized autoamtically on update

**ngx-graph version**
6.0.1

**Comment by ctaepper** [comment]
I experienced this too, but it was a problem on my side: the `nodes` input was an observable which hasn't emitted values at the time the chart rendered. If there aren't any nodes, the chart produces this error. Make sure, your nodes are populated and then optionally emit an update value

**Comment by doorman02** [comment]
Hi @ctaepper thanks I will check that. Nonetheless I think the chart should handle if there are no nodes. In my case my chart is bound to an array variable which initially is empty and is filled once the data has been received from a webservice. I could construct the graph only when there is data but it's feels more correct that the graph handles the scenario of not having any nodes in the array.

**Comment by KingDarBoja** [comment]
@doorman02 Can you provide more information about your issue? I am not sure how to reproduce this with the provided info. Do you have any stackblitz to look at?

Cheers!

**Comment by marjan-georgiev** [comment]
Merged #196, thanks all!
